### DISCUSSION TITLE
Get column (interval) width and space between columns.

### DISCUSSION THREAD

**Discussion by aldo-sanchez** [antvis/G2#5583] [Q&A]
G2 version: `5.0.24`

I'm building a small algorithm that groups x-axis labels. I'm using the axis `labelFormatter` to create a multi-level axis using `style` transforms that look more or less like this: 
```javascript
const xTranslation = header.shouldTranslateX ? '-25' : 0;
const style = {
  text: header.value,
  fontSize: 12,
  fontWeight: 400,
  fontFamily: normalTheme.typography.fontFamily,
  fill: normalTheme.colors.default.fg.onSubtle[5],
  fillOpacity: 0.5,
  textAlign: 'center',
  transform: `translate(${xTranslation}, ${yTranslation})`,
};
```

Where the`yTranslation` is based on whether or not my axis has a second (third, fourth, etc) level, and the `xTranslation` is used to determine whether or not I should move an axis value a little to the left, so that it lands between columns. Imagine the grouping has an even number of columns rather than odd. In this case the label should land between two columns.

Example of odd number of columns per group: two groups: g1 containing cols 1-3 and g2 containing cols 4-6. In our case g1 would land in col2 and g2 in col5.

Example of even number of columns per group: two groups: g1 containing cols 1-4 and g2 containing cols 5-8. In our case g1 should land between cols 2 and 3, and g2 between cols 6 and 7. 

Below I show real examples:

In the image below there are three levels. The first represents a date, the second a grouping of a quarter, and the third a grouping of either `freshman`, `junior`, or `senior`. The quarters need to be translated in `y` but not in `x`. While the `freshman | junior | senior` groupings need to be translated in both `x` and `y`.  For the most part my algorithm works well, and in 90% of my -25 translation is enough for the chart to look good.
<img width="810" alt="Screenshot 2023-09-21 at 20 58 15" src="https://github.com/antvis/G2/assets/22064588/3cb5310b-6d8e-4de0-909a-0a3b42d8c9b3">

However, when I have few bars, my -25 translation is not enough, see example below. In this case, the 2023 label is translated -25 but it is not enough to land between the first and second bar.
<img width="731" alt="Screenshot 2023-09-21 at 21 01 32" src="https://github.com/antvis/G2/assets/22064588/d7f35af5-497e-440e-8015-fccbdb5aee7d">

Is there a way to find the total width of each bar, and the space between them to make a better translation that will make my label land exactly between two bars? Is there a better way to do this in general?

Thanks!

**Comment by marcandrews** [comment]
A grouped axis label is just a link between groups with a label drawn below/besides the axis; no need to translate/shift labels.
